### PR TITLE
Feat/profile/age verification : Age verification in CreateProfile and EditProfile screens

### PR DESCRIPTION
**Pull request by charliemangano**
# Age verification in CreateProfile and EditProfile screens

## Description
This PR introduces age verification in the CreateProfile and EditProfile screens. It closes issue #314.

## Changes
I added an age condition in `UserViewModel`'s `dobValidators` list of validators. The minimum age is set to 16 and can be changed via the `MIN_AGE` constant. I added the corresponding error message and adapted the old all-encompassing one to be more precise for a malformatted date.

One the UI side, I added a simple text to the date of birth field to explain the restriction to users.

#### Added
None
#### Modified
- `app/src/main/java/com/android/periodpals/model/user/UserViewModel.kt`: added age validation logic and modified error messages
- `app/src/main/java/com/android/periodpals/ui/components/ProfileComponents.kt`: added explanatory text to date of birth field and changed the checks order to show proper error message
- `app/src/main/java/com/android/periodpals/resources/C.kt`: added test tag for explanatory text
- `app/src/test/java/com/android/periodpals/model/user/UserViewModelTest.kt`, `app/src/test/java/com/android/periodpals/ui/profile/CreateProfileTest.kt` and `app/src/test/java/com/android/periodpals/ui/profile/EditProfileTest.kt`: adapted and added tests
#### Removed
None

## Dependencies Added
None

## Testing
Adapted existing tests and added to check for the age validation components and behaviour.

## Screenshots
invalid date

too young

### CODE REVIEW

**Review by agonzalez-r — APPROVED** [review]
## Summary
LGTM!!! this implementation works great and it's an important addition to our app, thank you for this. 
Also, the age minimum, 16, was perfectly chosen since it is was is dictated by data protection laws in Europe (GDPR). So we shouldn't change that. 

## Important (which includes)
#### Code Quality
- Code is clean and consistent with form validation.
#### Functionality
- LGTM 
- minor remark: if you put a date into the future, the toast message displayed is that the user is not old enough. imo it should be date is not valid, but the important thing is that the user is not abel to crete the account with that DOB, so it's not important.

#### Security 
LGTM

## Testing
Tests added for the new feature are complete!

## Documentation
Complete! maybe add a comment for the min age constant, clarifying that it should be that by law.

**Review by francelu — APPROVED** [review]
## Summary
This PR adds validation to ensure that users are at least 16 years old, updates error messages for the date of birth field, and includes associated tests and UI changes to reflect the new validation requirements.

## Important
#### Code Quality
- The changes are well-organized and follow consistent code formatting.
- Constants and error messages are clearly defined and updated.
- The new validation function `isOldEnough` is well-integrated into the existing validation framework.

#### Functionality
- The new validation for the minimum age is correctly implemented using `LocalDate` and `DateTimeFormatter`.
- Edge cases for invalid dates and future dates are handled appropriately in the `isOldEnough` function.
- The UI correctly reflects the new validation requirement with a message indicating the minimum age.

#### Performance
- The changes should not have a significant impact on performance.

#### Security
- No security concerns identified.

## Testing
- The PR includes comprehensive tests for the new age validation.
- Tests verify that the validation returns the correct results for valid, recent, future, invalid, and empty dates.
- UI tests ensure that the minimum age message is displayed and that profiles cannot be saved if the user is too young.

## Documentation
- The code is well-commented, explaining the purpose of the new validation function.
- No additional documentation updates are necessary.

## Suggestions
- Consider adding a test for a boundary age case (e.g., exactly 16 years old) to ensure it is handled correctly.


## Steps before PR merged
- Add a test case for the boundary age of exactly 16 years old.

File: app/src/main/java/com/android/periodpals/model/user/UserViewModel.kt [hunk 1]
```diff
@@ -9,20 +9,24 @@ import com.dsc.form_builder.FormState
...
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 import java.util.Locale
 import kotlinx.coroutines.launch
 
 private const val TAG = "UserViewModel"
 
 private const val MAX_NAME_LENGTH = 128
 private const val MAX_DESCRIPTION_LENGTH = 512
+const val MAX_AGE = 16L
```
[the remark this answers is not in the document]
Comment (charliemangano) [reply app/src/main/java/com/android/periodpals/model/user/UserViewModel.kt:21]: Ah indeed, good catch!
